File: minecraft/origin/src/main/java/com/github/theword/queqiao/utils/LineProcessor.java
package com.github.theword.queqiao.utils;

import com.github.theword.queqiao.EventProcessor;
import com.github.theword.queqiao.config.Config;
import com.github.theword.queqiao.config.RegexConfig;
import lombok.RequiredArgsConstructor;

import java.util.regex.Matcher;

import static com.github.theword.queqiao.tool.utils.Tool.logger;

@RequiredArgsConstructor
public class LineProcessor {
    private final Config config;
    private final EventProcessor eventProcessor;

    public void processLine(String line) {
        logger.debug("LogWatcher: " + line);
        // 遍历并匹配聊天消息
        for (RegexConfig chatConfig : config.getChatRegexSet()) {
            Matcher chatMatcher = chatConfig.getPattern().matcher(line);
            if (chatMatcher.find()) {
                String playerName = chatMatcher.group(chatConfig.getPlayerGroup());
                String message = chatMatcher.group(chatConfig.getMessageGroup());

                // 检查聊天内容是否存在
                if (message != null && !message.isEmpty()) {
                    logger.info("[Chat] " + playerName + ": " + message);
                    eventProcessor.onPlayerChat(playerName, message);
                }
                return;
            }
        }

        // 遍历并匹配玩家加入消息
        for (RegexConfig joinConfig : config.getJoinRegexSet()) {
            Matcher joinMatcher = joinConfig.getPattern().matcher(line);
            if (joinMatcher.find()) {
                logger.info("[Join] " + line);
                String playerName = joinMatcher.group(joinConfig.getPlayerGroup());
                eventProcessor.onPlayerJoin(playerName);
                return;
            }
        }

        // 遍历并匹配玩家退出消息
        for (RegexConfig quitConfig : config.getQuitRegexSet()) {
            Matcher quitMatcher = quitConfig.getPattern().matcher(line);
            if (quitMatcher.find()) {
                logger.info("[Quit] " + line);
                String playerName = quitMatcher.group(quitConfig.getPlayerGroup());
                eventProcessor.onPlayerQuit(playerName);
                return;
            }
        }
    }
}

File: vanilla/origin/src/main/kotlin/com/github/theword/queqiao/utils/LineProcessor.kt
package com.github.theword.queqiao.utils

import com.github.theword.queqiao.EventProcessor
import com.github.theword.queqiao.config.Config
import org.slf4j.Logger

class LineProcessor(
    private val regexConfig: Config,
    private val config: com.github.theword.queqiao.tool.config.Config,
    private val logger: Logger,
    private val eventProcessor: EventProcessor
) {

    fun processLine(line: String) {
        logger.debug("LogWatcher: $line")
        if (config.subscribeEvent.isPlayerChat)
            regexConfig.chatRegexSet.forEach { chatConfig ->
                val chatMatcher = chatConfig.pattern.matcher(line)
                if (chatMatcher.find()) {
                    val playerName = chatMatcher.group(chatConfig.playerGroup)
                    val message = chatMatcher.group(chatConfig.messageGroup!!)
                    if (message != null && message.isNotEmpty()) {
                        logger.info("[Chat] $playerName: $message")
                        eventProcessor.onPlayerChat(playerName, message)
                    }
                    return
                }
            }

        if (config.subscribeEvent.isPlayerJoin)
            regexConfig.joinRegexSet.forEach { joinConfig ->
                val joinMatcher = joinConfig.pattern.matcher(line)
                if (joinMatcher.find()) {
                    logger.info("[Join] $line")
                    val playerName = joinMatcher.group(joinConfig.playerGroup)
                    eventProcessor.onPlayerJoin(playerName)
                    return
                }
            }

        if (config.subscribeEvent.isPlayerQuit)
            regexConfig.quitRegexSet.forEach { quitConfig ->
                val quitMatcher = quitConfig.pattern.matcher(line)
                if (quitMatcher.find()) {
                    logger.info("[Quit] $line")
                    val playerName = quitMatcher.group(quitConfig.playerGroup)
                    eventProcessor.onPlayerQuit(playerName)
                    return
                }
            }
    }
}